### ISSUE TITLE
[Feature Request] Add `val` / `var` / `static` / `global` coloration differences

### ISSUE THREAD

**Issue by Krutoy242** [raylras/zenscript-intelli-sense#99]
In previous versions `val` and `var` variables had a little bit different colors in editor that made them easily located.
Right now they all same colors.

![image](https://github.com/raylras/zenscript-intelli-sense/assets/936618/e85d1116-adf9-4f4c-a1df-84d2e0fabfcd)

Please, make `val` and `var` colors different again.

And additionally, I dont know it its possible, but would be great if statics and globals would have different colors too.

**Comment by raylras** [comment]
Fixed in c352df7f6c50cf7c85cf78fdf77168f6815c7714.

Furthermore, the Syntax Highlighting (implemented as the tmLanguage, or tml) isn't smart enough to recognize symbols in expressions like

![image](https://github.com/raylras/zenscript-intelli-sense/assets/61815515/ec0589f3-da63-46aa-a239-db858d744285)


The Semantic Highlighting API can make up for this, although it's a bit more complex.

For those interested, check out:
- https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide
- https://github.com/raylras/zenscript-intelli-sense/blob/ffe67a81dd8c19474a4ee32cbea967f511cd1182/zenscript-language-server/src/main/java/raylras/zen/lsp/provider/SemanticTokensProvider.java (not implemented as of this comment)

**Comment by raylras** [comment]
also, I cannot find the modifier to differentiate between `global` and `static` in Semantic Highlighting API
```ts
export enum SemanticTokenModifiers {
	declaration = 'declaration',
	definition = 'definition',
	readonly = 'readonly',
	static = 'static',
	deprecated = 'deprecated',
	abstract = 'abstract',
	async = 'async',
	modification = 'modification',
	documentation = 'documentation',
	defaultLibrary = 'defaultLibrary'
}
```
so, I don't know if it's possible.

[LSP 3.17 Specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens)

**Comment by Krutoy242** [comment]
Thank you for adding colorations to `val` at least!

**Comment by raylras** [comment]
It works!

![image](https://github.com/raylras/zenscript-intelli-sense/assets/61815515/26aa72b3-d750-4d01-ac54-6aac28f498fd)

but it's sloooooow in real projects, like E2EE

https://github.com/raylras/zenscript-intelli-sense/assets/61815515/55cbd583-4f57-4c15-a32e-7dd25b859c08

**Comment by Krutoy242** [comment]
If it takes 1 second I would like to have option for disabling this feature since it's too slow to actually use with pleasure.

**Comment by raylras** [comment]
Same operation, but much better now

```log
2023-12-27 20:34:45.157 [INFO] ZenLanguageService - didChange processUtils.zs started
2023-12-27 20:34:45.158 [INFO] ZenLanguageService - didChange processUtils.zs finished 1 ms
2023-12-27 20:34:45.207 [INFO] ZenLanguageService - semanticTokensFull processUtils.zs started
2023-12-27 20:34:45.208 [INFO] ZenLanguageService - semanticTokensFull processUtils.zs finished 1 ms
2023-12-27 20:34:45.255 [INFO] ZenLanguageService - documentSymbol processUtils.zs started
2023-12-27 20:34:45.256 [INFO] ZenLanguageService - documentSymbol processUtils.zs finished 0 ms
```

**Comment by raylras** [comment]
The relevant fixes have been released in v0.20.0, and now I am going to close this issue.
